app/services/induction/amend_participant_cohort.rb

```diff
@@ -147,11 +166,20 @@ def in_target_schedule?(induction_record)
 
     def participant_declarations
       return false unless participant_profile
+      return @participant_declarations if instance_variable_defined?(:@participant_declarations)
 
-      @participant_declarations ||= participant_profile
+      @participant_declarations = if cohort_changed_after_payments_frozen
```

Comment:
I think we're duplicating some of the existing eligibility checks in here; we have methods on `ParticipantProfile::ECF` that let you check if an ECT/mentor is eligible for transferring out of a frozen cohort to continue training, see:

https://github.com/DFE-Digital/early-careers-framework/blob/df43515d992b0e7c02754ae477d209ac6aefa6e4/app/models/participant_profile/ecf.rb#L95-L97

https://github.com/DFE-Digital/early-careers-framework/blob/df43515d992b0e7c02754ae477d209ac6aefa6e4/app/models/participant_profile/ect.rb#L31-L33

https://github.com/DFE-Digital/early-careers-framework/blob/df43515d992b0e7c02754ae477d209ac6aefa6e4/app/models/participant_profile/mentor.rb#L52-L54

These methods implement the logic Nathan provided to determine the outer pot of participants that are eligible:

```
1. The participant is in a payments frozen cohort
2. The profile does not have any completed declarations in the state eligible/payable/paid
3. The profile has at least 1 declaration of any other type that is eligible/payable/paid
4. The `induction_end_date` is `nil` for ECTs
5. The `mentor_completion_date` is `nil` for mentors
```

Reply:
We are already checking for 4/5 of that logic in different validations across the service.

I didn't want to call your method yet, because we need to check (as part of another ticket) the scheduled jobs we have running that check DQT for data updates and trigger cohort changes in participants. 
I suspect that for very rare cases those jobs might need to call this service to move 21 participants that meet all the 5 points above but the 3rd.

Let me try the other way around and call your method first and then undo the commit if later on those jobs need to do what I say.